### PR TITLE
Switching to assert_lint_err! macro #431

### PR DESCRIPTION
**Pull request by kamchy**
Refactored rules (#431)
- extracted rule name into CODE constant
- introduced enum value that displays proper message/hint
- used assert_lint_err! macro in one-line tests

### CODE REVIEW

File: src/rules/no_const_assign.rs
Comment (magurotuna) [inline src/rules/no_const_assign.rs:233]:
```suggestion
        col: 13,
```

Very small nit: extra whitespace here
(same for the other test cases)

File: src/rules/triple_slash_reference.rs
Comment (magurotuna) [inline src/rules/triple_slash_reference.rs:73]: Ooops I forgot to remove this test which was just for debugging. Thanks :D

**Comment by bartlomieju** [comment]
Thank you @kamchy! Before landing I'll wait for LGTM from @magurotuna

**Comment by kamchy** [comment]
@magurotuna - ping

File: src/rules/no_const_assign.rs
Comment (magurotuna) [inline src/rules/no_const_assign.rs:239]: I'd be happy if you could remove extra whitespaces after `col:` for all the test cases 👍

File: src/rules/no_const_assign.rs
Comment (magurotuna) [inline src/rules/no_const_assign.rs:293]: Let's replace `assert_lint_err_on_line` and `assert_lint_err_n` with `assert_lint_err!` macro as well. 
An example of how to replace is: https://github.com/denoland/deno_lint/blob/main/src/rules/getter_return.rs#L648-L664

File: src/rules/use_isnan.rs
Comment (magurotuna) [inline src/rules/use_isnan.rs:154]: Also replace it with the macro :)

File: src/rules/no_const_assign.rs
Comment (kamchy) [reply src/rules/no_const_assign.rs]: Fixed in a68ef06

File: src/rules/use_isnan.rs
Comment (kamchy) [reply src/rules/use_isnan.rs]: Done in 8e88eb6

File: src/rules/no_const_assign.rs
Comment (magurotuna) [reply src/rules/no_const_assign.rs]: Thanks! Looks like there are still undone cases. So can you please do the same for the following tests:

- `triple_slash_reference_invalid`
- `single_var_declarator_invalid`
- `no_throw_literal_invalid`
- `no_sparse_arrays_invalid`

File: src/rules/no_const_assign.rs
Comment (magurotuna) [reply src/rules/no_const_assign.rs]: great!

File: src/rules/use_isnan.rs
Comment (magurotuna) [reply src/rules/use_isnan.rs]: perfect 💯

**Comment by kamchy** [comment]
I tried to cleanup all rules so that only the macro assert_lint_err! is used. I also removed (commit 4698568) now-unused functions from test_util.rs. 

Please forgive me if I'm overusing the workflow by pushing to same PR several times...

**Comment by magurotuna** [comment]
@kamchy 

Never mind, thanks for addressing the review many times!
Overall looks great, but just one thing I would like you to do. See https://github.com/denoland/deno_lint/pull/684#discussion_r628235244. If it's done, I'm happy to approve.

File: src/rules/no_const_assign.rs
Comment (kamchy) [reply src/rules/no_const_assign.rs]: Done in dd6173c

**Comment by kamchy** [comment]
@magurotuna 
Fix for  https://github.com/denoland/deno_lint/pull/684#discussion_r628235244 done in dd6173c.  Please let me know if there's anything left in scope of #431. Thanks!

**Review by magurotuna — APPROVED** [review]
LGTM, awesome work!

**Review by bartlomieju — COMMENTED** [review]
LGTM, thank you @kamchy!

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**
